### PR TITLE
camerad: refactor exposure

### PR DESCRIPTION
**Pull request by deanlee**

### CODE REVIEW

File: selfdrive/camerad/cameras/camera_qcom2.cc [hunk 1]
```diff
@@ -1203,10 +1158,8 @@ void cameras_run(MultiCameraState *s) {
 
   LOG(" ************** STOPPING **************");
 
-  err = pthread_join(ae_thread_handle, NULL);
-  assert(err == 0);
+  for (auto &t : threads)
```
Comment (deanlee) [inline selfdrive/camerad/cameras/camera_qcom2.cc:1161]: wait all threads exit before call cameras_close

File: selfdrive/camerad/cameras/camera_qcom2.cc [hunk 2]
```diff
@@ -1203,10 +1158,8 @@ void cameras_run(MultiCameraState *s) {
...
+    t.join();
 
   cameras_close(s);
```
Comment (deanlee) [inline selfdrive/camerad/cameras/camera_qcom2.cc:1164]: wait all threads exit before call cameras_close

Comment (deanlee) [reply selfdrive/camerad/cameras/camera_qcom2.cc:1164]: submitted another PR to fix this problem https://github.com/commaai/openpilot/pull/19555

File: selfdrive/camerad/cameras/camera_common.cc [hunk 3]
```diff
@@ -361,6 +364,19 @@ void *processing_thread(MultiCameraState *cameras, const char *tname,
 
     callback(cameras, cs, cnt);
 
+#if defined(QCOM) || defined(QCOM2)
```
Comment (adeebshihadeh) [inline selfdrive/camerad/cameras/camera_common.cc:367]: if we start putting ifdefs like this everywhere, camera_common will just become what main.cc used to be

Comment (deanlee) [reply selfdrive/camerad/cameras/camera_common.cc:367]: yes you're right,I'll reconsider it.

Comment (deanlee) [reply selfdrive/camerad/cameras/camera_common.cc:367]: yes you're right,I'll reconsider it.

**Comment by deanlee** [comment]
use condition_variable to trigger camera_settings thread.it's more efficient and real-time as compared to sleep 50ms

**Pull request closed without merging**